moodle/Sniffs/Commenting/FileExpectedTagsSniff.php

```diff
@@ -0,0 +1,160 @@
+<?php
+
+// This file is part of Moodle - https://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <https://www.gnu.org/licenses/>.
+
+namespace MoodleHQ\MoodleCS\moodle\Sniffs\Commenting;
+
+use MoodleHQ\MoodleCS\moodle\Util\Docblocks;
+use MoodleHQ\MoodleCS\moodle\Util\TokenUtil;
+use PHP_CodeSniffer\Config;
+use PHP_CodeSniffer\Sniffs\Sniff;
+use PHP_CodeSniffer\Files\File;
+use PHPCSUtils\Tokens\Collections;
+
+/**
+ * Checks that a file has appropriate tags in either the file, or single artefact block.
+ *
+ * @copyright  2024 Andrew Lyons <andrew@nicols.co.uk>
+ * @license    https://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class FileExpectedTagsSniff implements Sniff
+{
+    /**
+     * The regular expression used to match the expected license.
+     *
+     * Note that the regular expression is applied using preg_quote to escape as required.
+     *
+     * Note that, if the regular expression is the empty string,
+     * then this Sniff will do nothing.
+     *
+     * Example values:
+     * - Empty string or null: No check is done.
+     *  ''
+     * - The GNU GPL v3 or later license with either http or https license text
+     *   '@https?://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later@':
+     *
+     * @var string
+     */
+    protected $preferredLicenseRegex = '@https?://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later@';
```

Comment:
Uhm, I think that this needs to be public if anybody wants to add a different license regex in their standard configuration. I may be wrong, but it sounds to me that I checked for examples when doing the TODO Sniff.

Reply:
Ah thanks. Fixed.